packages/core/schematics/ng-generate/control-flow-migration/types.ts

```diff
@@ -290,6 +297,24 @@ export class AnalyzedFile {
       analysis.ranges.push(range);
     }
   }
+
+  verifyCanRemoveImports() {
+    // if we already know it's not safe to remove the common module import
+    // skip this check entirely
+    if (this.removeCommonModule) {
+      const instances = lookupIdentifiersInSourceFile(this.sourceFile, 'CommonModule');
+      let count = 0;
+      for (let range of this.importRanges) {
+        for (let instance of instances) {
+          if (instance.getStart() >= range.start && instance.getEnd() <= range.end!) {
```

Comment:
Is it possible for the `range.end` to be `undefined` (looks like typing allow that)? Should we check for that explicitly?
```suggestion
          if (instance.getStart() >= range.start && instance.getEnd() <= range.end!) {
```

Reply:
In this case it's not possible for the range.end to be null. This check only happens on importRanges, which are guaranteed to have a range end.